src/components/themed/WalletListCurrencyRow.tsx

```diff
@@ -60,7 +63,14 @@ const WalletListCurrencyRowComponent = (props: Props) => {
   //
   const handlePress = useHandler(() => {
     triggerHaptic('impactLight')
-    if (onPress != null) onPress(wallet.id, tokenId, customAsset)
+    if (onPress != null) {
+      const maybePromise = onPress(wallet.id, tokenId, customAsset)
+      if (typeof (maybePromise as any)?.catch === 'function') {
+        ;(maybePromise as Promise<void>).catch((error: unknown) => {
+          showError(error)
+        })
+      }
+    }
```

Comment:
So for this, do:

```ts
const handlePress = useHandler(async () => {
  triggerHaptic('impactLight')
  if (onPress != null) await onPress(...)
}
```

This works because `await` handles promises & non-promises equally.